File: app/src/main/java/kagura/project/com/a8/association/semantique/SemantiqueMenu.java
package kagura.project.com.a8.association.semantique;

import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.ActivityInfo;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;

import kagura.project.com.a8.Encyclopedie;
import kagura.project.com.a8.R;

public class SemantiqueMenu extends AppCompatActivity {

    ImageView avatar;
    String avatarName;
    boolean isClickable = true;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);


        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);

        getWindow().requestFeature(Window.FEATURE_ACTION_BAR);
        getSupportActionBar().hide();
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);

        setContentView(R.layout.activity_association_menu);
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
        avatarName = preferences.getString(getString(R.string.avatar), null);

        avatar = (ImageView) findViewById(R.id.imageAvatar);

        switch(avatarName) {
            case "verMince":
                avatar.setBackgroundDrawable(getResources().getDrawable(R.drawable.ver_mince_hd_reverse));
                break;
            case "verGros":
                avatar.setBackgroundDrawable(getResources().getDrawable(R.drawable.ver_gros_hd));
                break;
        }

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        isClickable = true;
        super.onActivityResult(requestCode, resultCode, data);
        if(resultCode == 66){
            onBackPressed();
        }
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        overridePendingTransition(R.anim.right_start, R.anim.right_end);
    }

    public void startGame(View v) {
        if(isClickable){
            isClickable = false;
            int button = 1;
            switch (v.getId()){
                case R.id.level_1tv:
                    button = 1;
                    break;
                case R.id.level_2tv:
                    button = 2;
                    break;
                case R.id.level_3tv:
                    button = 3;
                    break;
            }
            Intent intentGame = new Intent(this, SemantiqueGame.class);
            intentGame.putExtra("level", button);
            this.startActivityForResult(intentGame, 0);
            overridePendingTransition(R.anim.left_start, R.anim.left_end);
        }
    }


    public void startEncyclopedie(View view) {
        if(isClickable){
            isClickable = false;
            Intent intentEncyclopedie = new Intent(this, Encyclopedie.class);
            this.startActivityForResult(intentEncyclopedie, 0);
            overridePendingTransition(R.anim.up_start, R.anim.up_end);
        }
    }

    public void back(View v) {
        onBackPressed();
    }
}

File: app/src/main/java/kagura/project/com/a8/association/semantique/SemantiqueMenu.kt
package kagura.project.com.a8.association.semantique

import android.annotation.SuppressLint
import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.view.View
import android.view.Window
import android.view.WindowManager
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceManager
import kagura.project.com.a8.Encyclopedie
import kagura.project.com.a8.R

class SemantiqueMenu : AppCompatActivity() {
    var avatar: ImageView? = null
    private var avatarName: String? = null
    private var isClickable = true
    //var fragmentResult: Fragment? = null
    //var fragmentManager: FragmentManager? = null

    @SuppressLint("SourceLockedOrientationActivity")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN)
        window.requestFeature(Window.FEATURE_ACTION_BAR)
        supportActionBar!!.hide()
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        setContentView(R.layout.activity_association_menu)
        val preferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        avatarName = preferences.getString(getString(R.string.avatar), null)
        avatar = findViewById<View>(R.id.imageAvatar) as ImageView
        when (avatarName) {
            "verMince" -> avatar!!.background = getDrawable(R.drawable.ver_mince_hd_reverse)
            "verGros" -> avatar!!.background = getDrawable(R.drawable.ver_gros_hd)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        isClickable = true
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == 66) {
            onBackPressed()
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        overridePendingTransition(R.anim.right_start, R.anim.right_end)
    }

    fun startEncyclopedie(@Suppress("UNUSED_PARAMETER") view: View?) {
        if (isClickable) {
            isClickable = false
            val intentEncyclopedie = Intent(this, Encyclopedie::class.java)
            this.startActivityForResult(intentEncyclopedie, 0)
            overridePendingTransition(R.anim.up_start, R.anim.up_end)
        }
    }

    fun back(@Suppress("UNUSED_PARAMETER") v: View?) {
        onBackPressed()
    }

    fun startGame(@Suppress("UNUSED_PARAMETER") v: View) {
        if (isClickable) {
            isClickable = false
            var button = 1
            when (v.id) {
                R.id.level_1tv -> button = 1
                R.id.level_2tv -> button = 2
                R.id.level_3tv -> button = 3
            }
            val intentGame = Intent(this, SemantiqueGame::class.java)
            intentGame.putExtra("level", button)
            this.startActivityForResult(intentGame, 0)
            overridePendingTransition(R.anim.left_start, R.anim.left_end)
        }
    }
}